### PR TITLE
TTD workaround for fs_stats issue.

### PR DESCRIPTION
**Pull request by mrkmarron**
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] documentation is changed or added

##### Affected core subsystem(s)

fs

##### Description of change
Fix for TTD break after change to FS Stats access.

PR-URL: https://github.com/mrkmarron/node-chakracore/commit/3e52041260d400589237192ebf8ca170dbac7041

### CODE REVIEW

File: src/node_file.cc [hunk 1]
```diff
@@ -1392,18 +1401,20 @@ static void Mkdtemp(const FunctionCallbackInfo<Value>& args) {
...
-  double* fields = env->fs_stats_field_array();
-  if (fields == nullptr) {
+  Local<Float64Array> fields = env->fs_stats_field_array();
+
+  if (fields.IsEmpty()) {
     // stat fields contains twice the number of entries because `fs.StatWatcher`
     // needs room to store data for *two* `fs.Stats` instances.
-    fields = new double[2 * 14];
+    Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(),
+        new double[2 * 14],
```
Comment (digitalinfinity) [inline src/node_file.cc:1410]: Who frees this allocation?

**Review by digitalinfinity — APPROVED** [review]
LGTM modulo that one question. Also, I'm guessing the perf impact of this change should be negligible but it would be good to run the benchmarks in benchmark\fs to verify.

File: src/node_file.cc (hunk 1 above)
Comment (kunalspathak) [reply src/node_file.cc:1410]: This is not related to @mrkmarron changes but was like this in upstream. I have [raised the question](https://github.com/nodejs/node/pull/11665/files#r107201164) to check this.

**Comment by kunalspathak** [comment]
@mrkmarron - I get your changes, but instead of modifying node code in so many places, why not have simple change to pass `statValues` from Jsland to C++ and use that `Float64Array` to extract `ArrayBuffer` and record in TTD?

Also, I agree with @digitalinfinity . We should do a perf run for `fs` module and compare the numbers with your change vs. before we got [this](https://github.com/nodejs/node-chakracore/pull/192/commits/6a5ab5d550719f416683ec0d588461b8bc9a8787) changes.

**Comment by digitalinfinity** [comment]
Landed in 87ce098 since TTD is broken badly without this change

**Comment by kunalspathak** [comment]
Sure, the main reason for doing minimal changes on node side is to avoid resolving merge conflicts around that code in future.

**Comment by mrkmarron** [comment]
I agree with the goal of minimal changes and I initially tried the approach we discussed of explicitly passing `statValues` from JSLand. However, I ended up needed to add this code in more places than I expected and also was seeing some odd replay divergences. So, I opted for this approach which turned out cleaner and simpler.

**Pull request merged**